### PR TITLE
19179 Added Amalgamating Businesses unit tests

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#19179

**This is only a subset of unit tests, but I think we can merge them in now and I'll create another PR for the others.**

*Description of changes:*
- app version = 5.6.40
- added element ids
- updated LEAR business duplicate check
- added foreign business duplicate check
- misc cleanup
- added test suite (WIP)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by leodube-aot — APPROVED** [review]
LGTM

File: src/components/Amalgamation/AmalgamatingBusinesses.vue [hunk 1]
```diff
@@ -410,20 +415,31 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
...
 
   /**
    * Check if business is already in table.
    * @param business The business being added.
    */
-  checkForDuplicateInTable (business: any): boolean {
-    const checkDuplication = this.getAmalgamatingBusinesses.find((b: any) =>
-      (b.type === AmlTypes.LEAR && b.identifier === business.businessInfo?.identifier) ||
+  private checkForDuplicateInTable (business: any): boolean {
+    const checkDuplication = this.getAmalgamatingBusinesses.find((b: AmalgamatingBusinessIF) =>
```
Comment (JazzarKarim) [inline src/components/Amalgamation/AmalgamatingBusinesses.vue:441]: 👍

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-626-evxarflh.web.app

File: src/components/Amalgamation/AmalgamatingBusinesses.vue [hunk 2]
```diff
@@ -410,20 +415,31 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
       corpNumber: this.corpNumber
     } as AmalgamatingBusinessIF
 
+    // Check for duplicate.
+    if (this.checkForDuplicateInTable(tingBusiness)) {
```
Comment (JazzarKarim) [inline src/components/Amalgamation/AmalgamatingBusinesses.vue:419]: Thanks for catching this Sev. I've tested in the temp URL, looks good!

**Review by JazzarKarim — APPROVED** [review]
Looks great!

**Pull request merged**